src/components/productsGrid/ProductsGrid.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Button,
+  LoadingMessage,
+  ErrorMessage,
+  HasNoResultsMessage,
+} from "../ui";
+import { ProductCard } from "./ProductCard";
+import { useProducts } from "../../hooks/useProducts";
+import { Product } from "../../types";
+import { useState } from "react";
+
+type ProductsGridProps = {
+  searchTerm: string;
+  sortBy: string;
+  sortOrder: string;
+};
+
+export const ProductsGrid = ({
+  searchTerm,
+  sortBy,
+  sortOrder,
+}: ProductsGridProps) => {
+  const [limit, setLimit] = useState(6);
+  const [isLoadingMore, setIsLoadingMore] = useState(false);
+  const {
+    data = [],
+    isLoading,
+    isError,
+  } = useProducts(searchTerm, sortBy, sortOrder, limit);
+
+  const hasResults = data.length > 0;
+
+  const incrementLimit = (increment: number) => {
+    setLimit((prev) => prev + increment);
+  };
+
+  const handleLoadMore = () => {
+    setIsLoadingMore(true);
```

Comment:
is you use isFetching you could delete all the logic of `setIsLoadingMore` and the timout